interfaces/builtin/serial_port.go

```diff
@@ -112,6 +112,11 @@ func (iface *serialPortInterface) SanitizeSlot(slot *interfaces.Slot) error {
 		if (usbProduct < 0x0) || (usbProduct > 0xFFFF) {
 			return fmt.Errorf("serial-port usb-product attribute not valid: %d", usbProduct)
 		}
+
+		usbInterfaceNumber, ok := slot.Attrs["usb-interface-number"].(int64)
+		if ok && (usbInterfaceNumber < 0x0) || (usbInterfaceNumber >= UsbMaxInterfaces) {
```

Comment:
I'd remove the parentheses around individual comparisons and instead put it around the `||` expression.
```
if ok && (usbInterfaceNumber < 0x0 || usbInterfaceNumber >= UsbMaxInterfaces) {
```
This way we know this triggers exactly when we want it to.

Reply:
Done!